lambda-code/reliability/src/lib/file_scanning.ts

```diff
@@ -0,0 +1,34 @@
+import { extractFileInputResponses } from "./dataLayer.js";
+import { getFileMetaData } from "./s3FileInput.js";
+import { FormSubmission } from "./types.js";
+
+export class FileScanningCompletionError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "FileScanningCompletionError";
+  }
+}
+
+export const verifyFileScanCompletion = async (formSubmission: FormSubmission) => {
+  const fileInputPaths = extractFileInputResponses(formSubmission);
```

Comment:
Probably not a big deal (especially on short submission)  but this function is going to be called twice every time the lambda code runs because both `sendToNotify` and `sendToVault` also call it internally.

Reply:
I know, but it was a way to short circuit further processes.  To only call it once we'd have to prop drill the file input paths down through the functions which seems overly complex to only run a function that iterates over an object.